### PR TITLE
Amdmigraphx update to rocm3.7

### PR DESCRIPTION
**Pull request by scxiao**
**Description**: Describe your changes.
   Update the AMDMIGraphX EP to use ROCm3.7, a newer release of the platform libs

**Motivation and Context**
  To get the AMDMIGraphX EP up to date  

- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc [hunk 1]
```diff
@@ -68,7 +68,7 @@ static void RegisterMIGraphXKernels(KernelRegistry& kernel_registry) {
   };
 
   for (auto& function_table_entry : function_table) {
-    kernel_registry.Register(function_table_entry());
+    auto status = kernel_registry.Register(function_table_entry());
```
Comment (jywu-msft) [inline onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc:71]: status isn't used?
or do something like
ORT_ENFORCE(kernel_registry.Register(function_table_entry()).IsOK());

Comment (scxiao) [reply onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc]: Thanks. Fixed it and just updated to the latest repo.

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**